### PR TITLE
Rfm69 merge into main

### PR DESCRIPTION
**Pull request by KISPE-Jamie**

### CODE REVIEW

File: low_level/comms.py [hunk 1]
```diff
@@ -24,14 +24,47 @@
 import serial
 import config
 from low_level.frameformat import PROTOCOL_DELIMITER, MAX_DATA_TYPES, DataType, DataTypeSize
+from config import OS, RaspberryPi
+try:
```
Comment (PaulMadle) [inline low_level/comms.py:28]: I would add commentary here to explain what is happening.

File: low_level/comms.py [hunk 2]
```diff
@@ -47,7 +80,35 @@ class CommsHandler:
...
         self.rx_state_machine = StateMachine()
-        self.comms = SerialComms(port, baud_rate)
+        if RaspberryPi is True:
+            self.radio = RadioComms(spi, CS, RESET, 434.0)
+        self.serial = SerialComms(config.COM_PORT, config.BAUD_RATE)
+
+    def change_comms(self):
+        if config.COMMS == "RF69":
+            return self.radio
+        elif config.COMMS == "SERIAL":
```
Comment (PaulMadle) [inline low_level/comms.py:90]: What happens if you select "SERIAL" on the RaspPi? Does it crash nastily?

File: main.py [hunk 3]
```diff
@@ -132,6 +133,9 @@ def __init__(self, *args, obj=None, **kwargs):
             tlm_slot["channel"].setText("")
             tlm_slot["value"].setText("")
 
+        if RaspberryPi is False:
+            self.comboBoxComms.setEnabled(False)
```
Comment (PaulMadle) [inline main.py:137]: Do we also need to disable if non-raspPi as well? i.e. default it to Serial and then disable selection?

File: main.py [hunk 4]
```diff
@@ -146,6 +150,11 @@ def __init__(self, *args, obj=None, **kwargs):
         config_register_callback(self.error_message_box)
         packet_init()
 
+        # Disable UART config if Linux TODO: Make this compatible, tty etc as well as COM
+        if OS == "Linux":
```
Comment (PaulMadle) [inline main.py:154]: I think this should be Raspberry Pi?

**Review by PaulMadle — COMMENTED** [review]
Hey Jamie, looks good on the whole... A few questions and observations.

File: main.py (hunk 4 above)
Comment (KISPE-Jamie) [reply main.py:154]: This can be deleted as there is a RaspberryPi check on line 136

File: low_level/comms.py (hunk 1 above)
Comment (PaulMadle) [reply low_level/comms.py:28]: Sounds good. I guess you should make the change and then I will close this comment.

File: low_level/comms.py (hunk 2 above)
Comment (PaulMadle) [reply low_level/comms.py]: Ok, can you show me it running on both the Pi and Windows so I can try and kill it?

File: main.py (hunk 4 above)
Comment (PaulMadle) [reply main.py]: ok, can I have a look at it running on RaspPi and Windows?

File: main.py (hunk 3 above)
Comment (PaulMadle) [reply main.py:137]: But what happens if you select serial on the RaspPi and try to run the code? Does it crash when you send something?

File: low_level/comms.py (hunk 1 above)
Comment (PaulMadle) [reply low_level/comms.py:28]: Comment added.

File: low_level/comms.py (hunk 2 above)
Comment (PaulMadle) [reply low_level/comms.py]: Demoed, all ok... Can now be resolved

File: main.py (hunk 4 above)
Comment (PaulMadle) [reply main.py]: Demo performed, all ok.

**Review by PaulMadle — APPROVED** [review]
Demoed on Rasp PI and Windows, small problem with digits of the year... Issue raised.

**Pull request merged**